api.planx.uk/proxy/ordnanceSurvey.ts

```diff
@@ -0,0 +1,52 @@
+import { useProxy } from './index';
+import { NextFunction, Request, Response } from "express";
+import { IncomingMessage } from 'http';
+
+export const OS_DOMAIN = "https://api.os.uk";
+
+const MAP_ALLOWLIST: RegExp[] = [
+  /http:\/\/(127\.0\.0\.1|localhost):(5173|7007)\//i, // Local development
+  /https:\/\/.*\.netlify\.app\//i, // Docs
+];
+
+export const useOrdnanceSurveyProxy = async (
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  if (!isValid(req)) return next({
+    status: 401,
+    message: "Unauthorised"
+  })
+
+  return useProxy({
+    target: OS_DOMAIN,
+    onProxyRes: (proxyRes, req) => setCORPHeaders(proxyRes, req),
+    pathRewrite: (fullPath, req) => appendAPIKey(fullPath, req)
+  })(req, res, next)
+};
+
+const isValid = (req: Request): boolean => isAllowListed(req) || isPlanX(req);
+
+const isAllowListed = (req: Request): boolean => MAP_ALLOWLIST.some(re => re.test(req.headers?.referer as string));
+
+const isPlanX = (req: Request): boolean => Boolean(req.headers.referer?.match(/^https:\/\/.*planx\.(pizza|dev|uk)\//i)?.length);
```

Comment:
I think we'll also need to account for custom domain names here (or in ALLOWLIST because I guess we'll actually want cross-origin CORS for that subset)?

Reply:
This is a great shout and something I just plain didn't consider 🙌  

I've gone for domains ending in `.gov.uk/` because...
 - No maintenance when we add new domains
 - Very few illicit calls to the API are likely to come from non-supported `.gov.uk` domains

I've also just simplified the `isValid()` functionality - one allow list works just fine, and adding "same-site" CORP headers really doesn't achieve anything if we're already properly validating.